pkg/getter/plugingetter.go

```diff
@@ -17,93 +17,132 @@ package getter
 
 import (
 	"bytes"
+	"context"
 	"fmt"
 	"os"
-	"path/filepath"
-	"strings"
 
 	"helm.sh/helm/v4/pkg/cli"
 	"helm.sh/helm/v4/pkg/plugin"
+	"helm.sh/helm/v4/pkg/plugin/schema"
 )
 
-// collectPlugins scans for getter plugins.
+// collectGetterPlugins scans for getter plugins.
 // This will load plugins according to the cli.
-func collectPlugins(settings *cli.EnvSettings) (Providers, error) {
-	dirs := filepath.SplitList(settings.PluginsDirectory)
-	descriptor := plugin.Descriptor{
-		Type: "download",
+func collectGetterPlugins(settings *cli.EnvSettings) (Providers, error) {
+
+	d := plugin.Descriptor{
+		Type: "getter/v1",
 	}
-	plugins, err := plugin.FindPlugins(dirs, descriptor)
+
+	plgs, err := plugin.FindPlugins([]string{settings.PluginsDirectory}, d)
 	if err != nil {
 		return nil, err
 	}
-	var result Providers
-	for _, p := range plugins {
-		// Get downloaders based on API version
-		var downloaders []plugin.Downloaders
-		if config, ok := p.Metadata().Config.(*plugin.ConfigDownload); ok {
-			downloaders = config.Downloaders
-		}
 
-		for _, downloader := range downloaders {
-			result = append(result, Provider{
-				Schemes: downloader.Protocols,
-				New: NewPluginGetter(
-					downloader.Command,
-					settings,
-					p.Metadata().Name,
-					p.Dir(),
-				),
-			})
+	pluginConstructorBuilder := func(plg plugin.Plugin) Constructor {
+		return func(option ...Option) (Getter, error) {
+
+			return &getterPlugin{
+				options: append([]Option{}, option...),
+				plg:     plg,
+			}, nil
 		}
 	}
-	return result, nil
-}
 
-// pluginGetter is a generic type to invoke custom downloaders,
-// implemented in plugins.
-type pluginGetter struct {
-	command  string
-	settings *cli.EnvSettings
-	name     string
-	base     string
-	opts     options
-}
+	results := make([]Provider, 0, len(plgs))
+
+	for _, plg := range plgs {
 
-func (p *pluginGetter) setupOptionsEnv(env []string) []string {
-	env = append(env, fmt.Sprintf("HELM_PLUGIN_USERNAME=%s", p.opts.username))
-	env = append(env, fmt.Sprintf("HELM_PLUGIN_PASSWORD=%s", p.opts.password))
-	env = append(env, fmt.Sprintf("HELM_PLUGIN_PASS_CREDENTIALS_ALL=%t", p.opts.passCredentialsAll))
-	return env
+		config := (plg.Metadata().Config).(*plugin.ConfigGetter)
+
+		results = append(results, Provider{
+			Schemes: config.Protocols,
+			New:     pluginConstructorBuilder(plg),
+		})
+	}
+	return results, nil
 }
 
-// Get runs downloader plugin command
-func (p *pluginGetter) Get(href string, options ...Option) (*bytes.Buffer, error) {
+func convertOptions(globalOptions, options []Option) (schema.GetterOptionsV1, error) {
```

Comment:
delete the plugin.ExecDownloader func in this PR, since you're removing the only call to it